### PR TITLE
Split TraceExporter class into ExportComponent and all the sub-components in an export package.

### PR DESCRIPTION
**Pull request by bogdandrutu**
PR got bigger, but the main changes are:

* Create a new package "export" where all the code related to exporting data was moved.
* Rename TraceExporter to ExportComponent.
* Extract inner classes SpanExporter and InProcessDebuggingHandler in normal classes to reduce complexity.

### CODE REVIEW

File: api/src/main/java/io/opencensus/trace/TraceExporter.java [hunk 1]
```diff
@@ -49,34 +50,66 @@ static TraceExporter getNoopTraceExporter() {
...
+       * immutable representation {@link SpanData}.
+       *
+       * <p>This may be called from a different thread than the one that called {@link Span#end()}.
+       *
+       * <p>Implementation SHOULD not block the calling thread. It should execute the export on a
+       * different thread if possible.
+       *
+       * @param spanDataList a list of {@code SpanData} objects to be exported.
+       */
+      public abstract void export(Collection<SpanData> spanDataList);
```
Comment (adriancole) [inline api/src/main/java/io/opencensus/trace/TraceExporter.java]: This is similar to zipkin's Reporter which is often async on this method.

File: api/src/main/java/io/opencensus/trace/TraceExporter.java [hunk 2]
```diff
@@ -49,34 +50,66 @@ static TraceExporter getNoopTraceExporter() {
...
 
   /**
-   * Unregisters the service handler with the provided name.
-   *
-   * @param name the name of the service handler that will be unregistered.
+   * A service that is used by the library to export {@code SpanData} for sampled spans (see
+   * {@link TraceOptions#isSampled()}).
    */
-  public abstract void unregisterServiceHandler(String name);
+  public abstract static class SampledSpansServiceExporter {
```
Comment (adriancole) [inline api/src/main/java/io/opencensus/trace/TraceExporter.java]: this name seems a bit long, and it isn't clear why the word Service is here at all.

File: api/src/main/java/io/opencensus/trace/TraceExporter.java (hunk 1 above)
Comment (adriancole) [inline api/src/main/java/io/opencensus/trace/TraceExporter.java]: maybe SpanExporter (if for some reason SpanReporter isn't acceptable)

Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/trace/TraceExporter.java]: Picked SpanExporter.

File: api/src/main/java/io/opencensus/trace/TraceExporter.java (hunk 2 above)
Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/trace/TraceExporter.java]: Fixed.

**Pull request merged**